test/Microsoft.AspNetCore.JsonPatch.Test/JsonPropertyTests.cs

```diff
@@ -0,0 +1,113 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Newtonsoft.Json;
+using System.Linq;
+using Xunit;
+
+namespace Microsoft.AspNetCore.JsonPatch.Test
+{
+    public class JsonPropertyTests
+    {
+        [Fact]
+        public void HonourJsonPropertyOnSerialization()
```

Comment:
We tend to name tests like `WhenIDoSomething_AnotherThingHappens_AndAnotherThing` - this would be like `Add_WithExpression_RespectsJsonPropertyName_ForModelProperty`